build_tools/github_actions/ci_build_llvm.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+# Copyright 2022 The StableHLO Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file is similar to build_mlir.sh, but passes different flags for
+# cacheing in GitHub Actions to improve build speeds.
+
+if [[ $# -ne 2 ]] ; then
+  echo "Usage: $0 <path/to/llvm> <build_dir>"
+  exit 1
+fi
+
+# LLVM source
+LLVM_SRC_DIR="$1"
+LLVM_BUILD_DIR="$2"
+
+# Check for LLD
+FLAGS=""
+LLD_FLAG="-DLLVM_ENABLE_LLD=ON"
+if command -v lld &> /dev/null
+then
+  echo "lld found, appending flag '$LLD_FLAG'"
+  FLAGS="$LLD_FLAG"
+else
+  echo "lld not found, using default linker"
+fi
```

Comment:
In what cases do we not have `lld` in our CI environment?

Reply:
Currently mac until I can figure out a good way to install lld on mac in the ci environment.

Reply:
What is the value that macOS support brings to the table? From what I can see, it complicates the configuration and the scripts, and it'll probably slow down CI overall, but I don't think I have a good understanding of what we're getting in return.

Reply:
Linux was giving trouble early on, mac was not. Initially I was using it to sanity check - i.e. if Linux failed but Mac passed, it probably had more to do with the GH Linux environment than the code change.

That said, I think that was largely due to the combination of caching and different experiments I was doing with config. I cleared all caches a little over a week ago and haven't had any Linux issues since then, so we're probably good to get rid of Mac.